Adapter/AdapterPattern.puml

```diff
@@ -0,0 +1,35 @@
+@startuml
+'https://plantuml.com/class-diagram
+
+class Client {
+    - adapter: Adapter
+}
+
+interface Adaptee {
+    + request()
+}
+
+class ConcreteAdaptee implements Adaptee {
+    + request()
+}
+
+interface Target {
+    + specificRequest()
+}
+
+interface Adapter {
+    - adaptee: Adaptee
```

Comment:
Java support field interface as constant and its initialization can be done on compile time only and we can't initilize it later. we can remove field here and add it aggregation type of ConcreteAdaptor class